stan/math/prim/scal/prob/neg_binomial_2_lpmf.hpp

```diff
@@ -79,33 +84,60 @@ return_type_t<T_location, T_precision> neg_binomial_2_lpmf(
   }
 
   for (size_t i = 0; i < max_size_seq_view; i++) {
-    if (include_summand<propto>::value) {
-      logp -= lgamma(n_vec[i] + 1.0);
-    }
-    if (include_summand<propto, T_precision>::value) {
-      logp += multiply_log(phi__[i], phi__[i]) - lgamma(phi__[i]);
-    }
-    if (include_summand<propto, T_location>::value) {
-      logp += multiply_log(n_vec[i], mu__[i]);
-    }
-    if (include_summand<propto, T_precision>::value) {
-      logp += lgamma(n_plus_phi[i]);
-    }
-    logp -= (n_plus_phi[i]) * log_mu_plus_phi[i];
-
-    // if phi is large we probably overflow, defer to Poisson:
-    if (phi__[i] > 1e5) {
-      logp = poisson_lpmf(n_vec[i], mu__[i]);
-    }
-
-    if (!is_constant_all<T_location>::value) {
-      ops_partials.edge1_.partials_[i]
-          += n_vec[i] / mu__[i] - (n_vec[i] + phi__[i]) / (mu__[i] + phi__[i]);
-    }
-    if (!is_constant_all<T_precision>::value) {
-      ops_partials.edge2_.partials_[i]
-          += 1.0 - n_plus_phi[i] / (mu__[i] + phi__[i]) + log_phi[i]
-             - log_mu_plus_phi[i] - digamma(phi__[i]) + digamma(n_plus_phi[i]);
+    if (phi__[i] > internal::neg_binomial_2_phi_cutoff) {
+      // Phi is large, deferring to Poisson.
```

Comment:
I kind of like more verbose comments (possibly because I am new to the codebase and had to learn a lot from gleaning at the code). Instead, I moved all the comments together to keep the code clean and tight while still letting people read more details about the code. Do you think that's better? (I don't insist on this in any way)

Reply:
Here's [Google style on implementation comments](https://google.github.io/styleguide/cppguide.html#Implementation_Comments).

It's no big deal either way.  The general rule for comments is to assume the reader of the code is better than the author at the language, so only non-idiomatic or non-obvious things get documented.  A good example is the comment about why there is cut-and-paste code.

The reason comments should be minimal is that it helps with code readability, which is the true goal.  That's largely because the code is the only thing you can ever really trust---comments can be obsolete or just plain wrong.  Too many comments get in the way of code readability, especially when they eat up a lot of vertical space and make it hard to take in the structure of the code at a glance.

P.S.  Also, that should be "delegate" instead of "defer";  my bad in just condensing rather than editing!

Reply:
I really want to keep the comments on how I got the formulae as I am confident about the code and not very confident about the math, so I want to "show my work" and make it easier for others to evaluate if it makes sense.